pkg/pulumiyaml/BUILD.bazel

```diff
@@ -0,0 +1,75 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "pulumiyaml",
+    srcs = [
+        "analyser.go",
```

Comment:
Is it possible to more loosely couple this? It seems strange to require this here. Or, if these files are auto generated (by Gazelle?): should they be committed or gitignored?

Reply:
Bazel does support `glob` so e.g. `srcs = glob(["**/*.go"])` could be used here. Currently these are I believe generated by Gazelle, so I expect it's a choice of "do we want to maintain `BUILD.bazel` files ourselves" (use `glob`, don't use Gazelle) or "do we want to try and pretend we're 'just writing Go' and use Gazelle to generate `BUILD.bazel` files".

Reply:
We could still use `glob`s, gazelle will actually support that just fine and not overwrite it, it's pretty smart that way!  Similarly, I had to adjust the `data` passed to tests in a few places.  So the build files are mostly auto-generated, but they need to be tweaked by hand, because we are not always following the standard convention, so gazelle can't generate that for you.

In a way I didn't bother with globs, exactly because I expect it's easiest to just use Gazelle to write these build files for us.  This is an extra command to run, but that seems like a worthy tradeoff for having faster and correct builds